### ISSUE TITLE
InvocationsPerInstance not published by async sagemaker endpoint

### ISSUE THREAD

**Issue by alexanderduben** [philschmid/terraform-aws-sagemaker-huggingface#9]
Hi, I took inspiration from your work and I am glad you've shared this example but I think the SageMakerVariantInvocationsPerInstance metric specification can't be used with async endpoints for autoscaling. It is stated in the official doc https://docs.aws.amazon.com/sagemaker/latest/dg/async-inference-monitor.html that InvocationsPerInstance metric is not published with async endpoints. I've tested the autoscaling with this metric and it really does not seem to be scaling at all. Some other custom metric supported by async endpoint should solve this issue.

**Comment by philschmid** [comment]
@alexanderduben You are correct for async it needs to be `ApproximateBacklogSizePerInstance`. I ll try to apply this soon. 
In the meantime you can check out this [example](https://github.com/huggingface/notebooks/blob/master/sagemaker/16_async_inference_hf_hub/sagemaker-notebook.ipynb) using the python-sagemaker-sdk to deploy. There the correct metric is used

**Comment by hingston** [comment]
@philschmid has there been any progress on this?

**Comment by philschmid** [comment]
No, not having time currently. Feel free to open a PR if you have.
